src/components/PeoplePage.tsx

```diff
@@ -1,29 +1,188 @@
+  return Math.ceil(year / 100);
+};
 
 export const PeoplePage = () => {
+  const [isLoading, setIsLoading] = useState(true);
+  const [people, setPeople] = useState<Person[]>([]);
+  const [error, setError] = useState<string | null>(null);
+  const { slug: selectedPersonSlug } = useParams();
+  const [searchParams, setSearchParams] = useSearchParams();
+  const sex = searchParams.get('sex');
+  const centuries = searchParams.getAll('centuries');
+  const query = searchParams.get('query') ?? '';
+  const sortColumn = searchParams.get('sort') as keyof Person | null;
+  const sortDirection = searchParams.get('order') === 'desc';
+
+  useEffect(() => {
+    getPeople()
+      .then(fetchedPeople => {
+        setPeople(
+          fetchedPeople.map(person => ({
+            ...person,
+            century: person.born ? getCentury(person.born) : undefined,
+          })),
+        );
+      })
+      .catch(() => {
+        setError('Something wrong');
+      })
+      .finally(() => {
+        setIsLoading(false);
+      });
+  }, []);
+
+  const filteredPeople = useMemo(() => {
+    const centuriesAsNumbers = centuries.map(Number);
+
+    // below const makes so that u chose all the existing sexes if u filter by sex or whatever
+    const filters: { condition: boolean; callback: (p: Person) => boolean }[] =
+      [
+        {
+          condition: !!sex,
+          callback: (person: Person) => person.sex === sex,
+        },
+        {
+          condition: centuries.length > 0,
+          callback: (person: Person) =>
+            person.century !== undefined &&
+            centuriesAsNumbers.includes(person.century),
+        },
+        {
+          condition: !!query,
+          callback: (person: Person) =>
+            query
+              ? person.name.toLowerCase().includes(query.toLowerCase())
+              : true,
+        },
+      ];
+
+    /*
+It starts with everyone in the list. For each filter rule, 
+if that rule is "turned on," it removes anyone who doesn't match. By the end, 
+you’re left with only the people who pass all the chosen filters.
+*/
+
+    return filters.reduce((acc, { condition, callback }) => {
+      if (condition) {
+        return acc.filter(callback);
+      }
+
+      return acc;
+    }, people);
+  }, [people, centuries, query, sex]);
+
+  const sortedFilteredPeople = useMemo(() => {
+    if (!sortColumn) {
+      return filteredPeople;
+    }
+    //
+
+    //here it does the filtering
+    const comparePeople = (a: Person, b: Person) => {
+      const valueA = a[sortColumn];
+      const valueB = b[sortColumn];
+
+      if (valueA === undefined && valueB === undefined) {
+        return 0;
+      }
+
+      if (valueA === undefined) {
+        return 1;
+      }
+
+      if (valueB === undefined) {
+        return -1;
+      }
+
+      if (sortDirection) {
+        return valueA > valueB ? -1 : valueA < valueB ? 1 : 0;
+      } else {
+        return valueA < valueB ? -1 : valueA < valueB ? 1 : 0;
```

Comment:
There seems to be a typo in the sorting logic. The condition `valueA < valueB ? -1 : valueA < valueB ? 1 : 0;` should likely be `valueA < valueB ? -1 : valueA > valueB ? 1 : 0;` to correctly handle ascending order sorting.